xbmc/network/GUIDialogNetworkSetup.h

```diff
@@ -25,17 +25,19 @@
 class CGUIDialogNetworkSetup : public CGUIDialogSettingsManualBase
 {
 public:
-  enum NET_PROTOCOL { NET_PROTOCOL_SMB = 0,
-                      NET_PROTOCOL_XBMSP,
-                      NET_PROTOCOL_FTP,
-                      NET_PROTOCOL_HTTP,
-                      NET_PROTOCOL_HTTPS,
-                      NET_PROTOCOL_DAV,
-                      NET_PROTOCOL_DAVS,
-                      NET_PROTOCOL_UPNP,
-                      NET_PROTOCOL_RSS,
-                      NET_PROTOCOL_SFTP,
-                      NET_PROTOCOL_NFS};
+  //! \brief A structure encapsulating properties of a supported protocol.
+  struct Protocol
+  {
+    bool paths;       //!< Protocol has path in addition to server name
+    bool usernames;   //!< Protocol uses logins
```

Comment:
Should we use singular for `paths` and `usernames` like you did for `password`, `port` etc?
Not sure if it makes sense to be more clear in case of the property names e.g. prefix with `support` .. `supportPath`, `supportUsername`, `supportBrowsing` etc.

Reply:
done.